frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/emoji_shortcut/emoji_picker_builder.dart

```diff
@@ -0,0 +1,367 @@
+      widget.onExit();
+
+      return KeyEventResult.handled;
+    } else if (event.logicalKey == LogicalKeyboardKey.space) {
+      if (_emojiController.text.isEmpty) {
+        widget.onExit();
+      }
+      return KeyEventResult.ignored;
+    } else if (event.logicalKey == LogicalKeyboardKey.backspace) {
+      if (_emojiController.text.isEmpty) {
+        widget.onExit();
+      } else {
+        _emojiController.text = _emojiController.text
+            .substring(0, _emojiController.text.length - 1);
+        _searchEmoji();
+      }
+      _deleteLastCharacters();
+      return KeyEventResult.handled;
+    } else if (event.character != null &&
+        !arrowKeys.contains(event.logicalKey)) {
+      _emojiController.text += event.character!;
+      _searchEmoji();
+      _insertText(event.character!);
+      return KeyEventResult.handled;
+    }
+
+    var newSelectedIndex = _selectedIndex;
+    if (event.logicalKey == LogicalKeyboardKey.arrowLeft) {
+      newSelectedIndex -= 1;
+    } else if (event.logicalKey == LogicalKeyboardKey.arrowRight) {
+      newSelectedIndex += 1;
+    } else if (event.logicalKey == LogicalKeyboardKey.arrowUp) {
+      newSelectedIndex -= 7;
+    } else if (event.logicalKey == LogicalKeyboardKey.arrowDown) {
+      newSelectedIndex += 7;
+    } else if (event.logicalKey == LogicalKeyboardKey.tab) {
+      newSelectedIndex += 7;
+      final currRow = (newSelectedIndex) % 7;
+      if (newSelectedIndex >= showingItems.length) {
+        newSelectedIndex = (currRow + 1) % 7;
+      }
+    }
+
+    if (newSelectedIndex != _selectedIndex) {
+      setState(() {
+        _selectedIndex = newSelectedIndex.clamp(0, showingItems.length - 1);
+      });
+      return KeyEventResult.handled;
+    }
+
+    return KeyEventResult.ignored;
+  }
+
+  _searchEmoji() {
+    final String query = _emojiController.text.toLowerCase();
+
+    searchEmojiList.emoji.clear();
+    for (final element in widget.state.categoryEmoji) {
+      searchEmojiList.emoji.addAll(
+        element.emoji.where((item) {
+          return item.name.toLowerCase().replaceAll(" ", "_").contains(query);
+        }).toList(),
+      );
+      searchEmojiList.emoji =
+          searchEmojiList.emoji.take(resultsFilterCount).toList();
+    }
+    setState(() {});
+  }
+
+  void _deleteLastCharacters({int length = 1}) {
+    final selection = widget.editorState.selection;
+    if (selection == null || !selection.isCollapsed) {
+      return;
+    }
+    final node = widget.editorState.getNodeAtPath(selection.end.path);
+    final delta = node?.delta;
+    if (node == null || delta == null) {
+      return;
+    }
+
+    // widget.onSelectionUpdate();
+    final transaction = widget.editorState.transaction
+      ..deleteText(
+        node,
+        selection.start.offset - length,
+        length,
+      );
+    widget.editorState.apply(transaction);
+  }
+
+  void _insertText(String text) {
```

Comment:
remove this function. use `widget.editorState.insertTextAtCurrentSelection(text);` instead